hal/src/tron/network/network.cpp

```diff
@@ -105,25 +107,13 @@ int if_init_platform(void*) {
     reserve_netif_index();
 
     /* en2 - Ethernet FeatherWing (optional) */
-    uint8_t mac[6] = {0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff};
-    {
-        // const uint32_t lsb = __builtin_bswap32(NRF_FICR->DEVICEADDR[0]);
-        // const uint32_t msb = NRF_FICR->DEVICEADDR[1] & 0xffff;
-        // memcpy(mac + 2, &lsb, sizeof(lsb));
-        // mac[0] = msb >> 8;
-        // mac[1] = msb;
-        /* Drop 'multicast' bit */
-        mac[0] &= 0b11111110;
-        /* Set 'locally administered' bit */
-        mac[0] |= 0b10;
-    }
+    uint8_t deviceId[HAL_DEVICE_ID_SIZE] = {};
+    hal_get_device_id(deviceId, sizeof(deviceId));
+    uint8_t* mac = deviceId + HAL_DEVICE_ID_SIZE - 6;
+    mac[5] += 3;
```

Comment:
Nit (can be done in a separate PR): let's add deviceid_hal_impl.h for all platforms (included from deviceid_hal.h, see other _impl.h headers as an example) and for rtl872x add a set of defines for MAC addresses to avoid having to use magic numbers:
```
HAL_DEVICE_MAC_WIFI_STA = 0
HAL_DEVICE_MAC_WIFI_AP  = 1
HAL_DEVICE_MAC_BLE = 2
HAL_DEVICE_MAC_ETHERNET = 3
```

Reply:
Let's leave it as-is and I'll get the deviceid hal refactored along with getting the BLE MAC address.

Reply:
Yup, let's do that :+1: